engine/player/unique_gear_shadowlands.cpp

```diff
@@ -4280,80 +4273,178 @@ void yasahm_the_riftbreaker( special_effect_t& effect )
 }
 
 // TODO: Add proc restrictions to match the weapons or expansion options.
-void cruciform_veinripper(special_effect_t& effect)
+void cruciform_veinripper( special_effect_t& effect )
 {
-
   struct sadistic_glee_t : public proc_spell_t
   {
     double scaled_dmg;
 
-    sadistic_glee_t(const special_effect_t& e)
-      : proc_spell_t("sadistic_glee", e.player, e.player->find_spell(353466), e.item),
-        scaled_dmg(e.driver()->effectN(1).average(e.item))
+    sadistic_glee_t( const special_effect_t& e )
+      : proc_spell_t( "sadistic_glee", e.player, e.player->find_spell( 353466 ), e.item ),
+        scaled_dmg( e.driver()->effectN( 1 ).average( e.item ) )
     {
       base_td = scaled_dmg;
     }
 
-    double base_ta(const action_state_t* /* s */) const override
+    double base_ta( const action_state_t* /* s */ ) const override
     {
       return scaled_dmg;
     }
   };
 
-  auto sadistic_glee = static_cast<sadistic_glee_t*>(effect.player->find_action("sadistic_glee"));
+  auto sadistic_glee = static_cast<sadistic_glee_t*>( effect.player->find_action( "sadistic_glee" ) );
 
-  if (!sadistic_glee)
-    effect.execute_action = create_proc_action<sadistic_glee_t>("sadistic_glee", effect);
+  if ( !sadistic_glee )
+    effect.execute_action = create_proc_action<sadistic_glee_t>( "sadistic_glee", effect );
   else
-    sadistic_glee->scaled_dmg += effect.driver()->effectN(1).average(effect.item);
+    sadistic_glee->scaled_dmg += effect.driver()->effectN( 1 ).average( effect.item );
 
   effect.spell_id = 357588;
   /* apparently due to proc rate being lower than reported in spell data (?) - emallson */
   effect.rppm_modifier_ = 0.5;
 
-
   /* override proc rate for tanks (40% of regular proc rate unless option is
      set), and allow override via expansion option */
   auto proc_option = effect.player->sim->shadowlands_opts.cruciform_veinripper_proc_rate;
-  if (proc_option == 0.0 && effect.player->position() == POSITION_FRONT) {
+  if ( proc_option == 0.0 && effect.player->position() == POSITION_FRONT )
+  {
     effect.rppm_modifier_ *= 0.4;
-  } else if(proc_option > 0.0) {
+  }
+  else if ( proc_option > 0.0 )
+  {
     effect.rppm_modifier_ *= proc_option;
   }
 
-  new dbc_proc_callback_t(effect.player, effect);
+  new dbc_proc_callback_t( effect.player, effect );
 }
 
+struct singularity_supreme_t : public stat_buff_t
```

Comment:
Because I clangformatted and made a mess, here's the start of the singularity supreme buff struct